src/OpenTelemetry.Instrumentation.Dependencies/Implementation/HttpWebRequestActivitySource.net461.cs

```diff
@@ -75,33 +76,20 @@ internal sealed class HttpWebRequestDiagnosticSource : DiagnosticListener
         private static Func<HttpWebResponse, bool> isWebSocketResponseAccessor;
         private static Func<HttpWebResponse, string> connectionGroupNameAccessor;
 
-        // Fields for controlling initialization of the HttpWebRequestDiagnosticSource singleton
-        private bool initialized = false;
-
-        private HttpWebRequestDiagnosticSource()
-            : base(DiagnosticListenerName)
-        {
-        }
-
-        public override IDisposable Subscribe(IObserver<KeyValuePair<string, object>> observer, Predicate<string> isEnabled)
-        {
-            IDisposable result = base.Subscribe(observer, isEnabled);
-            this.Initialize();
-            return result;
-        }
-
-        public override IDisposable Subscribe(IObserver<KeyValuePair<string, object>> observer, Func<string, object, object, bool> isEnabled)
+        internal HttpWebRequestActivitySource()
         {
-            IDisposable result = base.Subscribe(observer, isEnabled);
-            this.Initialize();
-            return result;
-        }
-
-        public override IDisposable Subscribe(IObserver<KeyValuePair<string, object>> observer)
-        {
-            IDisposable result = base.Subscribe(observer);
-            this.Initialize();
-            return result;
+            try
+            {
+                PrepareReflectionObjects();
+                PerformInjection();
+            }
+            catch (Exception ex)
+            {
+                // If anything went wrong, just no-op. Write an event so at least we can find out.
+                var activity = ActivitySource.StartActivity(InitializationFailedActivityName, ActivityKind.Server);
```

Comment:
This could be an abuse of activity - activity should represent an operation.. something with a begin and end notion.
Could we just log to EventSource at Error level instead?

Reply:
I updated to use `InstrumentationEventSource`. Please review and LMK if that works!